app/src/main/java/org/mozilla/fenix/settings/creditcards/CreditCardsSettingFragment.kt

```diff
@@ -23,10 +34,43 @@ import org.mozilla.fenix.settings.requirePreference
  */
 class CreditCardsSettingFragment : PreferenceFragmentCompat() {
 
+    private lateinit var creditCardsStore: CreditCardsFragmentStore
+    private var creditCardsAlreadyLoaded: Boolean = false
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        creditCardsStore = StoreProvider.get(this) {
+            CreditCardsFragmentStore(CreditCardsListState(creditCards = emptyList()))
+        }
+        loadCreditCards()
+    }
+
     override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
         setPreferencesFromResource(R.xml.credit_cards_preferences, rootKey)
     }
 
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        loadCreditCards()
```

Comment:
I don't fully understand a fragment lifecycle so this is more to help me satisfy my curious and help me understand this change. Can you explain why we need to `loadCreditCards()` here? Thanks

Reply:
Yes, when navigating away from this fragment to the _add card_ or _manage cards_ fragment, this fragment is not destroyed so when we come back from them `onCreate` is not called anymore. The view tough is destroyed, so when we come back `onCreateView` is still called. Given the fact that in the add/manage screens user can modify the card list by adding or deleting, we need to reload the card list and use it to update the state and implicitly the view. That is the reason we have `loadCreditCards` in both places, and the boolean to check we do not load the same list twice, but still call it as early as possible.